### DISCUSSION TITLE
Compilation issues

### DISCUSSION THREAD

**Discussion by melbnz** [EdenRidgway/XY-Wifi-Clock#13] [Q&A] [answered]
Hi, got a couple of minor challenges.  I'm not a competent programmer so struggling.  I believe i've got all libraries and set board correctly etc.  My issue is a couple of actions picked up during compilation that are not declared (i've tried in IDE 1.8 and 2.3.2).  Can anyone please point me in the right direction?  

I've been compiling XY-clock and can see some definitions in config manager but don't understand what to load when and how they all fit together!

Thanks :-)

C:\Users\melbu\Documents\Arduino\XY-Wifi-Clock\XY-Wifi-Clock.ino: In function 'void setup()':
C:\Users\melbu\Documents\Arduino\XY-Wifi-Clock\XY-Wifi-Clock.ino:341:9: error: 'loadSettings' was not declared in this scope; did you mean 'SPISettings'?
  341 |         loadSettings();
      |         ^~~~~~~~~~~~
      |         SPISettings
C:\Users\melbu\Documents\Arduino\XY-Wifi-Clock\XY-Wifi-Clock.ino:416:9: error: 'startWebServer' was not declared in this scope
  416 |         startWebServer();
      |         ^~~~~~~~~~~~~~

exit status 1

Compilation error: 'loadSettings' was not declared in this scope; did you mean 'SPISettings'?

**Comment by CasualCoder** [comment]
Hi @melbnz,
I have Arduino IDE v 1.8.19 and was able to compile the project without the errors you mention. Can you check and see if there could be any inadvertent changes to the XY-Wifi-Clock.ino file? Perhaps using a diff tool like Winmerge (Windows) or Meld (Linux).

If you wanna zip your project folder up and attach, happy to see if I can re-rpoduce the error also.

Best of luck!

**Comment by melbnz** [comment]
Thanks Eden, I really appreciate you responding 😊

 

I just can’t work out what I’m doing wrong!  And it will end up something silly!

 

Just upgraded to 1.8.19 and same compile error

 

I also made a new version of the ino copying RAW from your code and get same error

 

I cant see any parameters I need to edit – are there any?

 



 

Before you dive in…I did have a couple of library issues/differences

*	OTA – I already have the default ArduinoOTA.h loaded – no idea where it lives – but you use it with ESP examples.  When I added jandrassy version I get a conflict so moved his out of library folder
*	Wifimanager this is the version I installed which may be different?


 

Attached is my zipped project folder.

 

At some stage I’d like to understand what/how I could use the other ino files: config and website…

 

Thanks in advance

 

Mike

**Comment by CasualCoder** [comment]
Hi @melbnz ,

- I couldn't see your atatchment. Is it possible you forgot to attach or maybe gitHub dropped it for some reason?
- Don't think there are any parameters to edit.
- I'm using ArduinoOTA jandrassy version 1.0.9 ... what error do you get when trying to install it?
- Which version of WiFiManager are you using? I'm using WiFiManager by tablatronix (or is it by tzapu) version 2.0.16-rc.2 ... I just read this comment in the code XY-Wifi-Clock.ino line 3 ...  _"Need to install the WiFiManager library by tzapu (not the other ones)"_ ... so wonder if that is an issue?

p.s. I'm not Eden :-) ... they are the creator of this project. I'm just a humble user :-)

**Comment by melbnz** [comment]
Thanks Humble casual coder.  I replied via email and it excluded all my screenshots and attachments!  I’ve replied using GitHub this time.

I can install the jandrassy ota library but when I compile it states I have a conflict, “ Multiple libraries were found for "ArduinoOTA.h"
so I removed the new library (I can’t find the original but I used it for other esp ota projects.)

Added attachments 
[XY-Wifi-Clock.zip](https://github.com/EdenRidgway/XY-Wifi-Clock/files/14402477/XY-Wifi-Clock.zip)
![IMG_2996](https://github.com/EdenRidgway/XY-Wifi-Clock/assets/77428234/e2697237-fd43-42ed-9576-cb73237a7729)
![IMG_2995](https://github.com/EdenRidgway/XY-Wifi-Clock/assets/77428234/92b65e52-f207-409f-ac69-acb60f5f03c7)
 - hopefully…

thanks

**Reply by CasualCoder** [reply, marked as answer]
Thanks. Got your zip file now and the screenshots!

Think the issue is that you are missing some of the other '.ino' files, namely **ConfigManager.ino** and **WebSite.ino** ... these two files contain the declaration/implementations of the functions `loadSettings()` and `startWebServer()` respectively.

For reference, this is what my project's root folder looks like (I just cloned the whole repo):
![image](https://github.com/EdenRidgway/XY-Wifi-Clock/assets/3767445/20515958-14c6-41dc-ab9c-f42a04434325)

If I try and compile the project folder as you sent it, I get the same errors as you! (a good sign)
If I then copy over the missing files + folders into your project folder, then it compiles without issue.

Hope that helps.

p.s. Before copying across the missing files, after compilation, I get a warning that there are 2x arduinoOTA libraries available, and it has to chose one. This error goes away after copying missing files and compiling.
![image](https://github.com/EdenRidgway/XY-Wifi-Clock/assets/3767445/dcb09ff0-3e51-421d-9872-5dec374341b0)

**Reply by melbnz** [reply]
Thank you so much :-)  @CasualCoder I'm up and running after your 'education'

Really appreciate your support - thanks again

@EdenRidgway this is a great piece of code - I found it after buying the clock and deciding the sinilink app looked rather dodgy...  Love the degree of control and i will set an alarm sometime just to hear the Pink Panther...

**Reply by CasualCoder** [reply]
No worries @melbnz !!
You are well on your way to being a developer!
All the best!
